windows/vhba_hot_add_remove/vhba_test_prepare.yml

```diff
@@ -22,3 +22,37 @@
   when:
     - test_disk_controller_type == "nvme"
     - test_purpose == "hot-add"
+
+# For hot-add sata test, set the boot disk label firstly since boot order may be changed after hot-add disk.
+- name: "Set boot disk label"
+  block:
+    - name: "Get VM VirtualDisk info"
+      include_tasks: ../../common/vm_get_device_with_type.yml
+      vars:
+        device_vim_type: vim.vm.device.VirtualDisk
+
+    - name: "Set the disk label list"
+      ansible.builtin.set_fact:
+        boot_disk_label_list: "{{ boot_disk_label_list|default([]) + [item.deviceInfo.label] }}"
+      with_items: "{{ device_info_with_type }}"
+
+    - name: "Set the boot hard disk label when there's only one disk"
+      ansible.builtin.set_fact:
+        boot_disk_label: "{{ boot_disk_label_list[0] }}"
+      when: boot_disk_label_list | length == 1
+
+    - name: "Set the boot hard disk label when there are multiple disks"
+      ansible.builtin.set_fact:
+        boot_disk_label: "{{ 'Hard disk 1' if 'Hard disk 1' in boot_disk_label_list else boot_disk_label_list[0] }}"
+      when: boot_disk_label_list | length > 1
+
+    - name: "Display the boot hard disk label"
+      ansible.builtin.debug: var=boot_disk_label
+      when: boot_disk_label is defined
+  when:
+    - test_purpose == "hot-add"
+    - test_disk_controller_type == 'sata'
+    - (vm_firmware | lower == "bios")
+    - win_boot_disk_ctl_type is defined
+    - (win_boot_disk_ctl_type | lower not in ['ide', 'sata'])
+    - add_new_controller
```

Comment:
In `vhba_test.yml ` there already has 'add_new_controller' condition.

Reply:
It's for the second case. I.e add disk to existing controller.

Reply:
Sorry, I misunderstand it. Yes, we can remove it here.